src/kbmod_wf/task_impls/reproject_wu.py

```diff
@@ -12,3 +23,289 @@ def reproject_wu(input_wu=None, reprojected_wu=None, logger=None):
         f.write(f"Logged: {value} - {time.time()}\n")
 
     return reprojected_wu
+
+
+def reproject_wu(
+    original_wu_filepath: str = None,
+    uri_filepath: str = None,
+    reprojected_wu_filepath: str = None,
+    runtime_config: dict = {},
+    logger: Logger = None,
+):
+    wu_reprojector = WUReprojector(
+        original_wu_filepath=original_wu_filepath,
+        uri_filepath=uri_filepath,
+        reprojected_wu_filepath=reprojected_wu_filepath,
+        runtime_config=runtime_config,
+        logger=logger,
+    )
+
+    return wu_reprojector.reproject_workunit()
+
+
+class WUReprojector:
+    def __init__(
+        self,
+        original_wu_filepath: str = None,
+        uri_filepath: str = None,
+        reprojected_wu_filepath: str = None,
+        runtime_config: dict = {},
+        logger: Logger = None,
+    ):
+        self.original_wu_filepath = original_wu_filepath
+        self.uri_filepath = uri_filepath
+        self.reprojected_wu_filepath = reprojected_wu_filepath
+        self.runtime_config = runtime_config
+        self.logger = logger
+
+        self.overwrite = self.runtime_config.get("overwrite", False)
+        self.search_config = self.runtime_config.get("search_config", None)
+
+        # Default to 8 workers if not in the config. Value must be 0<num workers<65.
+        self.n_workers = np.max(1, np.min(self.runtime_config.get("n_workers", 8), 64))
+
+        # ! Decide if we want these from a runtime_config or exclusively read from the uri file
+        self.patch_corners = self.runtime_config.get("patch_corners", None)
+        self.guess_dist = self.runtime_config.get("guess_dist", None)
+        self.pixel_scale = self.runtime_config.get("pixel_scale", None)
+
+        # ! Decide if we want these from a runtime_config or calculate each time based on self.uri_params
+        self.image_width = self.runtime_config.get("image_width", None)
+        self.image_height = self.runtime_config.get("image_height", None)
```

Comment:
@wilsonbb there is a similar question here. The difference is that instead of either passing or reading from a file, we can either pass them or calculate them from the file. Again, it goes back to whether we'll still have .lst files that contain this info, or if we lose that with the work that Dino has done.

Reply:
I think here we should again presume calculating from the file, though that file format might end up changing from its current representation. But we can cross that road, once that's finalized

Reply:
Excellent. I agree, calculating from the file seems like the right choice here. I've removed these lines in favor of calculating them directly from the contents of the file.